R/zchunk_LA101.en_bal_IEA.R

```diff
@@ -39,72 +39,184 @@ module_energy_LA101.en_bal_IEA_DISABLED <- function(command, ...) {
+      mutate(value = value * conversion) %>%
+      group_by(GCAM_region_ID, sector, fuel, year) %>%
+      summarise(value = sum(value)) -> L101.en_bal_EJ_R_Si_Fi_Yh
+
+    # Setting to zero net fuel production from energy transformation sectors modeled under the industrial sector
+    # Setting to zero net production of fuels classified as coal at gas works (gas coke)
+    L101.en_bal_EJ_R_Si_Fi_Yh%>%
+      mutate(value = if_else(value <0 & grepl( "industry", sector), 0, value)) %>%
+      mutate(value = if_else(value <0 & sector == "in_gas processing", 0, value)) -> L101.en_bal_EJ_R_Si_Fi_Yh
+
+    # Create a template table with all applicable combinations of sector and fuel found in any region
+    # First, define the available combinations of sector and fuel and then repeat for all regions
+    L101.en_bal_EJ_R_Si_Fi_Yh %>% ungroup() %>%
+      select(sector, fuel) %>%
+      distinct(sector, fuel)%>%
+      repeat_add_columns(select(A_regions,GCAM_region_ID)) %>%
+      select(GCAM_region_ID, sector, fuel) -> L101.en_bal_EJ_R_Si_Fi_Yh_full
+
+    L101.en_bal_EJ_R_Si_Fi_Yh_full %>%
+      repeat_add_columns(tibble::tibble(year = HISTORICAL_YEARS)) %>%
+      left_join(select(L101.en_bal_EJ_R_Si_Fi_Yh,GCAM_region_ID, sector, fuel, year, value),
+                by = c("GCAM_region_ID","sector", "fuel","year")) %>%
+      mutate(value = if_else(is.na(value), 0, value)) -> L101.en_bal_EJ_R_Si_Fi_Yh_full
+
+
+
+    # Calculate the total primary energy supply (TPES) in each region as the sum of all flows that are inputs
+    # This guarantees that our TPES will be consistent with the tracked forms of consumption (i.e. no statdiffs, stock changes, transfers)
+    L101.en_bal_EJ_R_Si_Fi_Yh_full %>%
+      filter(grepl("in_", sector) | grepl("net_", sector)) %>%
+      mutate(sector = "TPES")%>%
+      group_by(GCAM_region_ID, sector, fuel, year) %>%
+      summarise(value = sum(value)) -> L101.in_EJ_R_TPES_Fi_Yh
+
+    # Append (rbind) TPES onto the end of the energy balances
+    L101.en_bal_EJ_R_Si_Fi_Yh_full %>%
+      bind_rows(L101.in_EJ_R_TPES_Fi_Yh) -> L101.en_bal_EJ_R_Si_Fi_Yh_full
+
+    # For downscaling of buildings and transportation energy, aggregate by fuel and country
+    # a: transport
+    L101.IEA_en_bal_ctry_hist_clean %>%
+      filter(grepl("trn", sector)) %>%
+      left_join_error_no_match(select(enduse_fuel_aggregation, fuel, trn), by ="fuel") %>%
+      select(-fuel, fuel=trn) -> L101.in_ktoe_ctry_trn_Fiea
+
+    L101.in_ktoe_ctry_trn_Fiea %>%
+      select(iso, sector, fuel, year, value, conversion) %>%
+      mutate(trn = value * conversion) %>%
+      group_by(iso, sector, fuel, year) %>%
+      summarise(value = sum(trn)) -> L101.in_EJ_ctry_trn_Fi_Yh
+
+    # b: buildings
+    L101.IEA_en_bal_ctry_hist_clean %>%
+      filter(grepl("bld", sector)) %>%
+      left_join_error_no_match(select(enduse_fuel_aggregation, fuel, bld), by ="fuel") %>%
+      select(-fuel, fuel=bld) -> L101.in_ktoe_ctry_bld_Fiea
```

Comment:
select(fuel = bld) did not work since it just selected bld labeled as fuel, not the entire tibble (which includes bld labeled as fuel)